### PR TITLE
Support FilePathsBase `AbstractPath` types

### LINKED ISSUE #56: Add support for FilePathsBase.jl
It would be useful to support the features of FilePathsBase.jl rather than requiring users to construct string paths.
```
julia> using FilePathsBase, JLSO

julia> x = p"foo"
p"foo"

julia> y = p"bar"
p"bar"

julia> p = x / y / "file.jlso"
p"foo/bar/file.jlso"

julia> JLSO.save(p, :baz=>1)
ERROR: MethodError: no method matching save(::PosixPath, ::Pair{Symbol,Int64})
Closest candidates are:
  save(::IO, ::Pair...; kwargs...) at /Users/maryjoramos/.julia/packages/JLSO/vWQhu/src/file_io.jl:76
  save(::IO, ::Any; kwargs...) at /Users/maryjoramos/.julia/packages/JLSO/vWQhu/src/file_io.jl:75
  save(::String, ::Any...; kwargs...) at /Users/maryjoramos/.julia/packages/JLSO/vWQhu/src/file_io.jl:77
Stacktrace:
 [1] top-level scope at REPL[26]:1
```

### PR DESCRIPTION
**Pull request by nickrobinson251**
- closes #56

### CODE REVIEW

File: test/file_io.jl [hunk 1]
```diff
@@ -148,14 +148,21 @@ end
...
-            single_item = JLSO.load("$path/$fmt-$k.jlso", k)
+            filepath = joinpath(dirpath, "$fmt-$k.jlso")
+            JLSO.save(filepath, k => v; format=fmt)
+            result = JLSO.load(filepath)
+            single_item = JLSO.load(filepath, k)
             @test result[k] == single_item[k] == v
         end
     end
+    @testset "String-type paths" begin
+        test_save_and_load(mktempdir())
```
Comment (nickrobinson251) [inline test/file_io.jl]: should these be the other way round (does it matter?)
```suggestion
        mktempdir(test_save_and_load())
```

File: src/file_io.jl [hunk 2]
```diff
@@ -74,7 +74,9 @@ Creates a JLSOFile with the specified data and kwargs and writes it back to the
 """
 save(io::IO, data; kwargs...) = write(io, JLSOFile(data; kwargs...))
 save(io::IO, data::Pair...; kwargs...) = save(io, Dict(data...); kwargs...)
-save(path::String, args...; kwargs...) = open(io -> save(io, args...; kwargs...), path, "w")
+function save(path::Union{AbstractPath, AbstractString}, args...; kwargs...)
```
Comment (rofinn) [inline src/file_io.jl]: FWIW, there's an `@compat` macro in FilePaths.jl that will create a wrapper string version for you.

https://github.com/rofinn/FilePaths.jl/blob/master/src/compat.jl

Downside is that it adds a couple extra dependencies.

File: test/file_io.jl (hunk 1 above)
Comment (nickrobinson251) [reply test/file_io.jl]: I think so? because then it clens up immediately?

File: test/file_io.jl [hunk 3]
```diff
@@ -148,14 +148,21 @@ end
...
+    end
+    @testset "Path-type paths" begin
+        test_save_and_load(mktempdir(SystemPath))
```
Comment (nickrobinson251) [inline test/file_io.jl]:
```suggestion
        mktempdir(test_save_and_load, SystemPath)
```

File: src/file_io.jl (hunk 2 above)
Comment (nickrobinson251) [reply src/file_io.jl]: yah, the extra dependencies is why i'm not using. If we needed many more changes like this and/or had more complex changes, like String/Path in many positions, then it _may_ be worth it. But i don't think it is worth it for just these two cases

**Pull request merged**